fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/MessageHelpers.java

```diff
@@ -579,5 +576,19 @@ public CoercionTrieNode(@Nullable final BiFunction<Descriptors.Descriptor, Objec
         public CoercionTrieNode getThis() {
             return this;
         }
+
+        @Override
+        public boolean equals(final Object other) {
+            if (this == other) {
+                return true;
+            }
+            if (!(other instanceof CoercionTrieNode)) {
```

Comment:
This is the only difference, right?

Reply:
Yes, correct, I refactored it accordingly.